docs/demos/demos/intro_demo.py

```diff
@@ -0,0 +1,256 @@
+    ops.Hadamard(q[0])
+    ops.CNOT(q[0], q[1])
+
+###############################################################################
+# We can now simulate the circuit, which will update its stored quantum state.
+
+simulate(circuit)
+
+###############################################################################
+# Printing the state reveals the expected state-vector $\frac{1}{\sqrt{2}}\left[1,
+# 0, 0, 1\right]$ corresponding to the state:
+#
+# $$\vert\psi\rangle = \frac{1}{\sqrt{2}}\left(\vert00\rangle + \vert11\rangle\right)$$
+
+circuit.state
+
+###############################################################################
+# ## Measurements
+#
+# Measurements work like any other operation in dwave-gate. The main difference is that the operation generates a measurement value when simulated which can be stored in the classical register by piping it into a classical bit.
+#
+# We can reuse the circuit from above by simply unlocking it and appending a `Measurement` to it.
+
+circuit.unlock()
+with circuit.context as (q, c):
+    m = ops.Measurement(q[1]) | c[0]
+
+###############################################################################
+# :::note
+# We stored the measurement instance as `m`, which we can use for post-processing. It's also possible to do this with all other operations in the same way, allowing for multiple identical operation applications.
+# ```python
+# with circuit.context as q, _:
+#     single_x_op = ops.X(q[0])
+#     # apply the X-gate again to the second qubit
+#     # using the previously stored operation
+#     single_x_op(q[1])
+# ```
+# This procedure can also be shortened into a single line for further convience.
+# ```python
+# ops.CNOT(q[0], q[1])(q[1], q[2])(q[2], q[3])
+# ```
+# :::
+
+###############################################################################
+# The circuit should now contain 3 operations: a Hadamard, a CNOT and a measurment.
+
+print(circuit)
+
+###############################################################################
+# When simulating this circuit, the measurement will be applied and the measured value will be stored in the classical register. Since a measurement will affect the quantum state, the resulting state will have collapsed into the expected result dependent on the value which has been measured.
+
+simulate(circuit)
+
+###############################################################################
+# If the measurement result is 0 the state should collapse into $\vert00\rangle$, and if the measurement result is 1 the state should collapse into $\vert11\rangle$. Outputting the measurement value and the state reveals that this is indeed the case.
+
+print(circuit.bits[0].value)
+print(circuit.state)
+
+###############################################################################
+# ## Measurement post-access
+# Since we stored the measurement operation in `m`, we can use it to access the state as it was before the measurement.
+#
+# :::note
```

Comment:
Not rendering
![image](https://github.com/dwavesystems/dwave-gate/assets/34041130/e3cbc0d9-5ac5-4687-8b9e-81ea7258d68b)